pkg/apis/v1alpha3/types.go

```diff
@@ -49,12 +52,14 @@ const (
 
 // LogStatus defines the current status of the log resource.
 type LogStatus struct {
-	Path string `json:"path,omitempty"`
-	Size int64  `json:"size"`
+	Path            string `json:"path,omitempty"`
+	Size            int64  `json:"size"`
+	IsStored        bool   `json:"isStored"`
+	ErrorOnStoreMsg string `json:"errorOnStoreMsg"`
```

Comment:
yeah per my comment https://github.com/tektoncd/results/pull/704/files#r1630064603 @khrm lets add an `errorOnStoreCode int` as well

@avinal @sayan-biswas fyi

Reply:
@khrm consider the old prototype for using this change over in https://github.com/tektoncd/results/pull/713

I want that call in the watcher's dynamic.go to not just check if it is stored, and if it is not, requeue to consider pruning later.

I want it to also see that if it is not stored per the update made in pkg/api/server/v1alpha2/logs.go 's `handleReturn` (where it has the original "error"), but because of an error on the api server side, to see if the particular error is one of the grpc intermittent sort of hiccups that mean we can re-attempt the log storage.

In this way, we finally address the lack of retry on error that @sayan-biswas @avinal @enarha and myself have lamented for months.

Now, if there is some flaw in the way I've connected the dots here, then let's you @khrm, myself, @avinal @sayan-biswas @enarha discuss either here on in the next WG call.

As a backup, I'm also fine with setting a boolean "IsRetrablyErrorOccurred" field. in `LogStatus`, have the handleReturn code make the decision if the error that occured in an intermittent, retryable sort of thing, and then the dynamic.go code works off of that to try the `UpdateLog` call again.

thanks

Reply:
I have added a new field for this.

Reply:
LGTM @khrm thanks

@enarha @avinal @sayan-biswas as I am not in the OWNERS file anywhere, we need reviews from you all to get this finally merged. 

PTAL